### ISSUE TITLE
[Bug]: fluent PrimaryButton color contrast issue in Fluent 2 DarkTheme

### ISSUE THREAD

**Issue by ishwins** [microsoft/fluentui#28455]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 129.34 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Safari: 16.5.1
  npmPackages:
    @fluentui/fluent2-theme: ^8.107.0 => 8.107.26
    @fluentui/font-icons-mdl2: ^8.2.0 => 8.5.13
    @fluentui/jest-serializer-merge-styles: ^8.0.12 => 8.0.24
    @fluentui/merge-styles: ^8.5.7 => 8.5.7
    @fluentui/react: ^8.106.1 => 8.110.7
    @fluentui/react-icons: ^2.0.188 => 2.0.204
    @fluentui/react-theme: ^9.1.1 => 9.1.7
    @fluentui/scheme-utilities: ^8.3.0 => 8.3.26
    @fluentui/style-utilities: ^8.8.1 => 8.9.6
    @fluentui/theme: ^2.4.2 => 2.6.25
    @fluentui/utilities: ^8.13.6 => 8.13.18
```

### Are you reporting Accessibility issue?

yes

### Reproduction

This is for fluent 2 theme
https://developer.microsoft.com/en-us/fluentui#/controls/web/button

## Actual Behavior
fluent PrimaryButton in Fluent 2 DarkTheme doesn't meet contrast ratio.

## Expected Behavior
Fluent Primary Button in Fluent 2 DarkTheme meets contrast ratio. The button background color should be #0F6CBD (same as fluent 2 light theme, as its the only component that keeps the same color values across themes)

<img width="705" alt="Screenshot 2023-07-06 at 4 14 36 PM" src="https://github.com/microsoft/fluentui/assets/138827407/5471261b-7619-462e-94be-c5c7d41017ba">

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
